nemo/collections/llm/quantization/quantizer.py

```diff
@@ -0,0 +1,332 @@
+    The process consist of several steps:
+
+        1. Loading a Nemo model from disk using appropriate parallelism strategy
+        2. Calibrating the model to obtain appropriate algorithm-specific scaling factors
+        3. Producing output directory
+
+    The output directory produced is intended to be consumed by TensorRT-LLM toolbox
+    for efficient inference. This can be achieved using NeMo inference containers.
+
+    Available quantization methods are listed in `QUANT_CFG_CHOICES` dictionary above.
+    Please consult Model Optimizer documentation https://nvidia.github.io/TensorRT-Model-Optimizer/ for details.
+
+    Quantization algorithm can also be conveniently set to None to perform only weights export step
+    for TensorRT-LLM deployment. This is useful to getting baseline results for a full-precision model.
+    """
+
+    def __init__(self, quantization_config: dict, export_config: dict):
+        """Initialize Quantizer with quantization and export configurations.
+
+        Expected keys in `quantization_config`:
+            - algorithm: (optional) str
+            - awq_block_size: int (only for awq algorithms)
+            - sq_alpha: float (only for smooth quant algorithms)
+            - enable_kv_cache: bool (default: None i.e. auto-detect based on algorithm and decoder_type)
+
+        Expected keys in `export_config`:
+            - dtype: str/int
+            - decoder_type: (optional) str
+            - inference_tensor_parallel: int
+            - inference_pipeline_parallel: int
+            - path: str
+        """
+        if not HAVE_MODELOPT:
+            raise RuntimeError("nvidia-modelopt is needed to use Quantizer") from HAVE_MODELOPT_ERROR
+        if not torch.cuda.is_available():
+            raise EnvironmentError("GPU is required for the quantization.")
+
+        self.quantization_config = quantization_config
+        self.export_config = export_config
+        self.nemo_checkpoint_path = None
+
+        algorithm = quantization_config.get("algorithm", None)
+        dtype = export_config["dtype"]
+
+        # Quantization sanity checks
+        assert algorithm is None or algorithm in QUANT_CFG_CHOICES, f"Unsupported quantization algorithm: {algorithm}"
+        # Export sanity checks
+        if export_config is not None:
+            assert dtype in SUPPORTED_DTYPE, f"Unsupported export dtype: {dtype}"
+        self.torch_dtype = torch_dtype_from_precision(dtype)
+
+    def load_quantizable_model(self, nemo_checkpoint_path: str, calib_tp: int = 1, calib_pp: int = 1) -> llm.GPTModel:
+        trainer = nl.Trainer(
+            devices=calib_tp * calib_pp,
+            strategy=nl.MegatronStrategy(
+                tensor_model_parallel_size=calib_tp,
+                pipeline_model_parallel_size=calib_pp,
+            ),
+            plugins=nl.MegatronMixedPrecision(precision='16-mixed'),
+        )
+        fabric = trainer.to_fabric()
+        trainer.strategy.setup_environment()
```

Comment:
I think `fabric.launch()` would be better

Reply:
Changed